### PR TITLE
Fix the go module errors due to version conflicts of k8s

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR does

- Replace k8s.io/apiextensions-apiserver and k8s.io/apimachinery modules with v0.18.6
- Add go mod tidy prechecks

### Why we need it

- Error occurred while I ran `go mod tidy`.
```go
➜  ks-devops git:(feat/jenkins-pipelinerun) go mod tidy                          
go: finding module for package github.com/googleapis/gnostic/openapiv2
kubesphere.io/devops/controllers/devopscredential imports
        k8s.io/client-go/tools/record imports
        k8s.io/apimachinery/pkg/util/strategicpatch tested by
        k8s.io/apimachinery/pkg/util/strategicpatch.test imports
        k8s.io/apimachinery/pkg/util/strategicpatch/testing imports
        github.com/googleapis/gnostic/openapiv2: module github.com/googleapis/gnostic@latest found (v0.5.5, replaced by github.com/googleapis/gnostic@v0.4.0), but does not contain package github.com/googleapis/gnostic/openapiv2
```

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -241,7 +241,22 @@ jobs:
     name: Lint
     runs-on: ubuntu-20.04
     steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v2.1.3
+        with:
+          go-version: 1.15
```
Comment (LinuxSuRen) [inline .github/workflows/build.yaml:247]: We should keep using the same version of go. See also https://github.com/kubesphere/ks-devops/blob/6d9f1f47378be638360bad0f23177dcb409beaa2/go.mod#L3

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Thanks for the fixing. I left some comments. Please take a look at it.

/remove-label do-not-merge/work-in-progress

File: .github/workflows/build.yaml (hunk 1 above)
Comment (JohnNiang) [reply .github/workflows/build.yaml:247]: Got it. But how about this one?

https://github.com/kubesphere/ks-devops/blob/a4598c1bbc47d2e4b95d2a99a4f193ad9aad2e2c/.github/workflows/build.yaml#L217-L226

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -241,7 +241,22 @@ jobs:
...
       - uses: actions/checkout@v2
+      # See also https://github.com/GoogleCloudPlatform/golang-samples/blob/78dfa41f10b449ba7a06d9793cbd81878d44a4fb/.github/workflows/go.yaml#L29-L53
+      - name: Run go mod tidy on all modules
+        run: find . -name go.mod -execdir go mod tidy \;
```
[the remark this answers is not in the document]
Comment (JohnNiang) [reply .github/workflows/build.yaml:251]: Alright. I will refine the command.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (LinuxSuRen) [reply .github/workflows/build.yaml:247]: That is not a reason to have multiple Golang versions. In another word, it's incorrect.

Comment (JohnNiang) [reply .github/workflows/build.yaml]: I will fix the wrong Golang version in another pull request.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Pull request merged**